ignite/src/test/java/org/hibernate/ogm/datastore/ignite/test/cfg/LongIndexNameWithIndexAnnotationTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.ignite.test.cfg;
+
+import org.apache.ignite.cache.QueryIndex;
+import org.hibernate.ogm.OgmSession;
+import org.hibernate.ogm.datastore.ignite.utils.IgniteTestHelper;
+import org.hibernate.ogm.utils.OgmTestCase;
+import org.junit.Test;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Index;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.fest.assertions.Assertions.assertThat;
+
+/**
+ * Entities with long names that results in an index with greater than 255 chars is invalid in Ignite.
+ * This ensures we raise an error and provide useful information to the user
+ */
+public class LongIndexNameWithIndexAnnotationTest extends OgmTestCase {
+
+	@Test
+	public void testLongEntityIndexName() throws Exception {
+		try ( OgmSession session = openSession() ) {
+			Set<QueryIndex> indexes = IgniteTestHelper.getIndexes( session.getSessionFactory(), EntityWithCustomIndex.class );
+			assertThat( indexes.size() ).isEqualTo( 1 );
+			assertThat( indexes.iterator().next().getName() ).isEqualToIgnoringCase( "SimpleIndexName" );
```

Comment:
Nice!

You could make this test even better using:
`assertThat( indexes ).onProperty( "name" ).containsOnly( "SimpleIndexName" )`

Reply:
I think we need some additional test cases:

1. What happens when the index name looks like something like :
    ```
    Table(indexes = {@Index(name = "Idx @#\\@    weird chars", columnList = "id")})
     ```
    Is that even possible?
2. What happens if there are two indexes created on different fields in the same class?
3. What happens if two indexes have the same name except for the the case? Example: "IndexNAME" and "Indexname"
4. What happens when the index span multiple columns? Will it work using `@Index( columnList = "field1,field2")`?

This is just a question because I'm not familiar with it: Can we support the `unique` property in the `@Index` annotation?